src/Stage/ReadStageInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/*
+ *
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Stage;
+
+/**
+ * Retrieves data from the applicable data provider and sets it as a request parameter called data.
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+interface ReadStageInterface
+{
+    public const OPERATION_ATTRIBUTE_KEY = 'read';
+
+    /**
+     * @return object|iterable|null
+     */
+    public function apply(array $attributes, array $parameters, ?array $filters, string $method, array $normalizationContext);
```

Comment:
`apply` isn't very descriptive (I'm not a fond of "stage" too btw :D). Can't we more explicit names?

Reply:
Which one do you have in mind? Something like `read`?
I know you would say that :smile: Naming is really hard and subjective. "Stage" was interesting because it was not used and its meaning was alright: it's a "step in a process".

Reply:
Stage sounds like “util” or “service” to me: too generic and not very descriptive

Reply:
But it is something kind of generic. It's a step or an operation. How would you call the `read`, `serialize`, `validate`, etc. things?

Reply:
Make it a callable? `__invoke`?

Reply:
Good idea! WDYT @dunglas?